gammapy/analysis/core.py

```diff
@@ -296,7 +299,11 @@ def _spectrum_extraction(self):
             exclusion_region = Map.read(datasets_settings.background.exclusion)
             bkg_maker_config["exclusion_mask"] = exclusion_region
         bkg_maker = ReflectedRegionsBackgroundMaker(**bkg_maker_config)
-        safe_mask_maker = SafeMaskMaker(methods=["aeff-default", "aeff-max"])
+
+
+        safe_mask_selection = self.config.datasets.safe_mask.methods
+        safe_mask_settings = self.config.datasets.safe_mask.settings.dict()
+        safe_mask_maker = SafeMaskMaker(methods=safe_mask_selection, **safe_mask_settings)
```

Comment:
Maybe we could have the `safe_mask_maker` as a property of `analysis` and avoid  code duplication or at least pass it as an argument to `_spectrum_extraction` and `_map_making` internal in the `get_datasets` method.